src/app/clusters/service-area-server/service-area-server.h

```diff
@@ -0,0 +1,751 @@
+                               const DataModel::Nullable<AreaTypeTag>     & aAreaType,
+                               const DataModel::Nullable<LandmarkTag>     & aLandmarkTag,
+                               const DataModel::Nullable<PositionTag>     & aPositionTag,
+                               const DataModel::Nullable<FloorSurfaceTag> & aSurfaceTag  );
+
+    /**
+     * @brief  Modify/replace an existing location in the supported locations list
+     * @param aLocationId unique identifier of this location
+     * @param aMapId identifier of supported map
+     * @param aLocationName human readable name for this location (empty string if not used)
+     * @param aFloorNumber represents floor level - negative values for below ground
+     * @param aAreaType common namespace Area tag - indicates an association of the location with an indoor or outdoor area of a home
+     * @param aLandmarkTag common namespace Landmak tag - indicates an association of the location with a home landmark
+     * @param aPositionTag common namespace Position tag - indicates the position of the location with respect to the landmark
+     * @param aSurfaceTag common namespace Floor Surface tag - indicates an association of the location with a surface type
+     * @return true if the location is a member of supported locations, the modifications pass all validation checks and the location was modified 
+     * @note caller is responsible for change notification
+     * @note if aLocationName is larger than kLocationtNameMaxSize, it will be truncated
+
+     */
+    bool ModifySupportedLocation( uint32_t                                     aLocationId, 
+                                  const DataModel::Nullable<uint8_t>         & aMapId, 
+                                  const CharSpan                             & aLocationName,
+                                  const DataModel::Nullable<int16_t>         & aFloorNumber,
+                                  const DataModel::Nullable<AreaTypeTag>     & aAreaType,
+                                  const DataModel::Nullable<LandmarkTag>     & aLandmarkTag,
+                                  const DataModel::Nullable<PositionTag>     & aPositionTag,
+                                  const DataModel::Nullable<FloorSurfaceTag> & aSurfaceTag  );
+
+    /**
+     * @brief When a supported location changes or is deleted, update other attributes as required
+     * @param aLocationId unique identifier of this location
+     * 
+     * @note Any selected locations that depend on the modified supported location will be deleted.
+     * @note If the device is operating and selected locations becomes null, the device shall stop operation.
+     * @note Any progress elements that depend in the modified supported location will be deleted.
+     * @note if the updated location is the current location, the current location will be set to null.
+     * @note MATTER change notifications are made for any attributes that change
```

Comment:
Most of the simple attribute functions don't handle notifications. Some of the more complex ones do.
Extensive commenting in the doxygen, but could still be confusing.

Reply:
All of the attribute modifiers should now submit the appropriate notifications.